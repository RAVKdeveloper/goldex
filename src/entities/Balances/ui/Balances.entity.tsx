import { FC } from 'react'

import { BalanceCard } from 'goldex-ui-library'

import { useTranslation } from 'react-i18next'

import { useBalanceInfoQuery } from '@/shared/service/hooks'

import s from '../assets/style.module.css'

export const Balances: FC = () => {
  const { t } = useTranslation()
  const { data } = useBalanceInfoQuery()

  return (
    <div data-testid='balances' className={s.root}>
      {data &&
        data.data &&
        data.data.length > 0 &&
        data.data.map(({ balance, type }) => (
          <div key={type} className={s.container}>
            {type === 'profit' ? (
              <BalanceCard currency={t('profit')} balance={balance} height='lg' />
            ) : (
              <BalanceCard currency={type} balance={balance} height='lg' />
            )}
          </div>
        ))}
    </div>
  )
}
