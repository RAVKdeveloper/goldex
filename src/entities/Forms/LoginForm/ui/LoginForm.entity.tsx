import { FC } from 'react'
import { Button, TextField } from 'goldex-ui-library'
import { EyeIcon } from 'goldex-icon-library'

import s from '../assets/style.module.css'

export const LoginFormEntity: FC = () => {
  return (
    <form className={s.form}>
      <div className={s.header}>
        <h1 className={s.title}>Вход</h1>
        <p className={s.supportingText}>Для продолжения войдите в свою учетную запись</p>
      </div>
      <div className={s.content}>
        <TextField variant='input' label='Логин' />
        <TextField
          type='password'
          variant='input'
          label='Пароль'
          icon={<EyeIcon color='#FFFFFF' opacity={0.4} width={20} height={20} />}
        />
      </div>
      <Button variant='primary' size='full' disabled>
        Войти
      </Button>
    </form>
  )
}
