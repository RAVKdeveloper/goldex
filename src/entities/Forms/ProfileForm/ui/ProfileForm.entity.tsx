import { FC } from 'react'
import { Button, TextField, UploadAvatar } from 'goldex-ui-library'

import { useTranslation } from 'react-i18next'

import s from '../assets/style.module.css'

export const ProfileFormEntity: FC = () => {
  const { t } = useTranslation()

  return (
    <form className={s.form}>
      <div className={s.header}>
        <h2 className={s.title}>{t('profileTitle')}</h2>
        <p className={s.supportingText}>{t('profileSupText')}</p>
      </div>
      <div className={s.content}>
        <UploadAvatar size='lg' />
        <TextField label={t('labels.username')} sizeType='lg' />
      </div>
      <div className={s.button}>
        <Button variant='primary' size='es' disabled>
          {t('btn.save')}
        </Button>
      </div>
    </form>
  )
}
