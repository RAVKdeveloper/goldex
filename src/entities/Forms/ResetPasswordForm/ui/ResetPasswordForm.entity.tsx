import { FC } from 'react'
import { Button, TextField } from 'goldex-ui-library'

import { useTranslation } from 'react-i18next'

import s from '../assets/style.module.css'

export const ResetPasswordFormEntity: FC = () => {
  const { t } = useTranslation()

  return (
    <form className={s.form}>
      <div className={s.header}>
        <h2 className={s.title}>{t('resetPasswordTitle')}</h2>
        <p className={s.supportingText}>{t('resetPasswordSupText')}</p>
      </div>
      <div className={s.content}>
        <TextField label={t('labels.currentPassword')} sizeType='lg' />
        <TextField label={t('labels.newPassword')} sizeType='lg' />
        <TextField label={t('labels.confirmCurrentPassword')} sizeType='lg' />
      </div>
      <div className={s.button}>
        <Button variant='primary' size='es' disabled>
          {t('btn.save')}
        </Button>
      </div>
    </form>
  )
}
