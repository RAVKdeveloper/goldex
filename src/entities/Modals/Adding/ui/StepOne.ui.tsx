import { FC } from 'react'

import { TextField, ModalActions } from 'goldex-ui-library'

import { useTranslation } from 'react-i18next'

import { useAddingBalance } from '../logic/useAddingBalance.model'

import s from '../assets/style.module.css'

export const StepOneAddingModal: FC = () => {
  const { t } = useTranslation()
  const { canselAdding, nextStep, changeInput } = useAddingBalance()

  return (
    <article className={s.stepOne}>
      <TextField
        label={t('labels.currensy')}
        variant='select'
        placeholder={t('placeholders.noselected')}
      />
      <TextField label={t('labels.amount')} name='amount' onChange={changeInput} />
      <ModalActions
        btnTextCancel={t('btn.cansel')}
        btnTextAction={t('btn.adding')}
        cbAction={nextStep}
        cbCansel={canselAdding}
      />
      <input hidden type='submit'></input>
    </article>
  )
}
