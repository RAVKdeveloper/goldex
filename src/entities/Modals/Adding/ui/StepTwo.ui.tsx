import React, { FC } from 'react'

import { ModalActions, TextField } from 'goldex-ui-library'

import { useTranslation } from 'react-i18next'

import { useAddingBalance } from '../logic/useAddingBalance.model'

import s from '../assets/style.module.css'

export const StepTwoAddingModal: FC = React.forwardRef(() => {
  const { t } = useTranslation()
  const { canselAdding, changeInput, submit, amount } = useAddingBalance()

  return (
    <article className={s.stepOne}>
      <TextField label={t('labels.address')} name='addres' onChange={changeInput} />
      <div className={s.column}>
        <p className={s.currency}>
          <span>{t('labels.currensy')}</span> 2
        </p>
        <p className={s.amount}>
          <span>{t('labels.amount')}</span> {Number(amount).toLocaleString('en')}
        </p>
      </div>
      <ModalActions
        btnTextCancel={t('btn.cansel')}
        btnTextAction={t('btn.iSent')}
        cbAction={submit}
        cbCansel={canselAdding}
      />
    </article>
  )
})
