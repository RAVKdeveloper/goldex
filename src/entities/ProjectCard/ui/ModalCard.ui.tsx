import { FC } from 'react'

import { useTranslation } from 'react-i18next'

import s from '../assets/style.module.css'

interface Props {
  isOpen: boolean
  isBlocked: boolean
  clickToBlocked: () => void
}

export const ModalProjectCardUi: FC<Props> = ({ isBlocked, isOpen, clickToBlocked }) => {
  const { t } = useTranslation()

  return (
    <article className={`${s.modal} ${isOpen && s.open}`}>
      <ul>
        {/* Сдесь должны быть новые иконки */}
        <li className={s.item}>{t('modal.projectCard.key')}</li>
        <li onClick={clickToBlocked} className={`${s.item} ${s.red}`}>
          {isBlocked ? t('modal.projectCard.unblocked') : t('modal.projectCard.blocked')}
        </li>
      </ul>
    </article>
  )
}
