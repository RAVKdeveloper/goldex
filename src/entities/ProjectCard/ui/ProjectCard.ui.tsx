import { FC } from 'react'

import { ProjectCard } from 'goldex-ui-library'

import { useProjectCard } from '../logic/useProjectCard.model'
import { ModalProjectCardUi } from './ModalCard.ui'

import s from '../assets/style.module.css'

export const ProjectCardEntity: FC = () => {
  const { open, isOpen, isBlocked, blocked } = useProjectCard()

  return (
    <div className={s.root}>
      <ProjectCard
        name='IJDJI'
        status='active'
        turnover={'35353'}
        minus={-324}
        plus={3243}
        textStatus='eokeot'
        textTurnover='etet'
        cbOptions={open}
      />
      <ModalProjectCardUi isOpen={isOpen} isBlocked={isBlocked} clickToBlocked={blocked} />
    </div>
  )
}
