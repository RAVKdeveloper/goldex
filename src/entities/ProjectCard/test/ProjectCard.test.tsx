import { render, screen } from '@testing-library/react'

import { ProjectCardEntity } from '../ui/ProjectCard.ui'

describe('testing ProjectCardEntity component', () => {
  it('render', () => {
    const component = render(<ProjectCardEntity />)

    const rootComponent = screen.getByTestId('projectCard')

    expect(rootComponent).toBeInTheDocument()
    expect(component).toMatchSnapshot()
  })
})
