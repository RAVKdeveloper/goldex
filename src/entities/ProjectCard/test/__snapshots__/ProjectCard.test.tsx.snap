// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing ProjectCardEntity component render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="root"
        data-testid="projectCard"
      >
        <projectcard
          minus="-324"
          name="IJDJI"
          plus="3243"
          status="active"
          textstatus="eokeot"
          textturnover="etet"
          turnover="35353"
        />
        <article
          class="modal false"
        >
          <ul>
            <li
              class="item"
            >
              modal.projectCard.key
            </li>
            <li
              class="item red"
            >
              modal.projectCard.blocked
            </li>
          </ul>
        </article>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="root"
      data-testid="projectCard"
    >
      <projectcard
        minus="-324"
        name="IJDJI"
        plus="3243"
        status="active"
        textstatus="eokeot"
        textturnover="etet"
        turnover="35353"
      />
      <article
        class="modal false"
      >
        <ul>
          <li
            class="item"
          >
            modal.projectCard.key
          </li>
          <li
            class="item red"
          >
            modal.projectCard.blocked
          </li>
        </ul>
      </article>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
